pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -239,6 +239,10 @@ func (m *kubeGenericRuntimeManager) startContainer(ctx context.Context, podSandb
 		}
 	}
 
+	if restartCount > 0 {
+		m.recordContainerEvent(pod, container, "", v1.EventTypeNormal, events.ContainerDied, fmt.Sprintf("Container died. Restarting container %s", container.Name))
```

Comment:
This is a little bit strange to see `ContainerDied` event in the `startContainer` function. Is there a better place to do this, e.g. when container actually dies?

Reply:
It was previously named as _containerRestarted_ and during review event reason was changed. There can definitely be some other place where containerDied event is recorded. But as community is still contemplating whether  adding this event message is needed or not, its better to first confirm to work on anything further.

Reply:
I'll add the change and update here.